BoxContentSDK/BoxContentSDK/Views/BOXUserAvatarImageView.m

```diff
@@ -0,0 +1,265 @@
+    BOXCircularAvatarBackgroundShadeCount,
+} BOXCircularAvatarBackgroundShade;
+
+@interface BOXUserAvatarImageView()
+@property (nonatomic, readwrite, strong) BOXUserAvatarRequest *request;
+@end
+
+@implementation BOXUserAvatarImageView
+
+- (instancetype)initWithFrame:(CGRect)frame
+                         user:(BOXUserMini *)user
+                       client:(BOXContentClient *)client
+{
+    if (self = [super initWithFrame:frame]) {
+        _user = user;
+        _client = client;
+        
+        self.contentMode = UIViewContentModeScaleAspectFill;
+        self.layer.masksToBounds = YES;
+
+        if (user) {
+            [self render];
+        }
+    }
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame
+                       client:(BOXContentClient *)client
+{
+    self = [self initWithFrame:frame user:nil client:nil];
+    return self;
+}
+
+- (void)layoutSubviews
+{
+    [super layoutSubviews];
+    self.layer.cornerRadius = self.bounds.size.width / 2;
+}
+
+- (void)setUser:(BOXUserMini *)user
+{
+    NSString *oldUserID = _user.modelID;
+    _user = user;
+    if (![oldUserID isEqualToString:user.modelID]) {
+        [self render];
+    }
+}
+
+#pragma mark - Private helpers
+
+- (void)render
+{
+    [self.request cancel];
+    self.request = [self.client userAvatarRequestWithID:self.user.modelID type:BOXAvatarTypeLarge];
+    id<BOXContentCacheClientProtocol> cacheClient = [self.client cacheClient];
+    if ([cacheClient respondsToSelector:@selector(retrieveCacheForUserAvatarRequest:completion:)]) {
+        [cacheClient retrieveCacheForUserAvatarRequest:self.request
+                                            completion:^(UIImage *image, NSError *error) {
+                                                if (image && !error) {
+                                                    [self renderImage:image];
+                                                } else {
+                                                    [self renderNameBasedAvatar];
+                                                    [self renderRealUserAvatarFromRemote];
+                                                }
+                                            }];
+    } else {
+        [self renderNameBasedAvatar];
+        [self renderRealUserAvatarFromRemote];
+    }
+}
+
+- (void)renderNameBasedAvatar
+{
+    UIImage *image = [self nameBasedAvatarForUser:self.user size:self.bounds.size];
+    [self renderImage:image];
+}
+
+- (void)renderRealUserAvatarFromRemote
+{
+    [self.request performRequestWithProgress:nil completion:^(UIImage *image, NSError *error) {
+        if (image && !error) {
+            [self renderImage:image];
+        } else {
+            [self renderNameBasedAvatar];
+        }
+    }];
+}
+
+- (void)renderImage:(UIImage *)image
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        self.image = image;
+        [self setNeedsLayout];
+    });
+}
+
+#pragma mark - Helpers for name-based avatar
```

Comment:
Sourced from https://git.dev.box.net/Mobile/box-ios-share-sdk-source/blob/dev/BoxShareSDK/BoxShareSDK/Categories/UIImage%2BBOXCircularAvatar.m

Reply:
wonder if we could have this new class BOXUserAvatarImageView.m in ShareSDK instead? we want to avoid bringing UI components to content sdk, which would be imported as the base networking layer.

Reply:
@boxtcarpel and I had some discussions about where to land this, and the problem with sticking it into something like the share SDK is that we'll need to make other components like the Comments SDK dependent on it. Long term, I think we'll merge all the UI SDKs, but until then this seemed like the best place to put it. Happy to discuss further though.